code/modules/mob/living/carbon/human/species/wryn.dm

```diff
@@ -49,6 +49,106 @@
 	//Default styles for created mobs.
 	default_hair = "Antennae"
 
+	var/datum/action/innate/wryn_sting/wryn_sting
+	button_on = FALSE
+
+/datum/species/wryn/on_species_gain(mob/living/carbon/human/H)
+	..()
+	wryn_sting = new
+	wryn_sting.Grant(H)
+
+/datum/species/wryn/on_species_loss(mob/living/carbon/human/H)
+	..()
+	if(wryn_sting)
+		wryn_sting.Remove(H)
+
+/* Wryn Sting Action Begin */
+
+//Define the Sting Action
+/datum/action/innate/wryn_sting
+	name = "Wryn Sting"
+	desc = "Readies Wryn Sting for stinging."
+	button_icon_state = "wryn_sting_off"		//Default Button State
+	check_flags = AB_CHECK_LYING | AB_CHECK_CONSCIOUS | AB_CHECK_STUNNED
+
+//What happens when you click the Button?
+/datum/action/innate/wryn_sting/Trigger()
```

Comment:
Styling suggestion for this proc, use an early return format. Example:
```dm
/datum/action/innate/wryn_sting/Trigger()
	if(!..())
		return

	var/mob/living/carbon/user = owner

	if(user.restrained() || user.pulledby || user.buckled) //Is your Wryn restrained, pulled, or buckled? No stinging
		to_chat(user, "<span class='notice'>You need freedom of movement to sting someone!</span>")
		return

	if(user.wear_suit) //Is your Wryn wearing a Hardsuit or a Laboat that's blocking their Stinger?
		to_chat(user, "<span class='notice'>You must remove your hardsuit, labcoat, or jacket before using your Wryn stinger.</span>")
		return

	if(user.getStaminaLoss() <= 50)	//Does your Wryn have enough Stamina to sting?
		user.dna.species.button_on = TRUE
		UpdateButtonIcon()
		user.dna.species.select_target(user)
	else
		to_chat(user, "<span class='notice'>You feel too tired to use your Wryn Stinger at the moment.</span>")
		return
```

Reply:
Buh. That makes it look so much nicer!!

Reply:
Did this and a little more. 
`if(user.getStaminaLoss() >= 50)	//Does your Wryn have enough Stamina to sting?
		to_chat(user, "<span class='notice'>You feel too tired to use your Wryn Stinger at the moment.</span>")
 else 
                user.dna.species.button_on = TRUE
		UpdateButtonIcon()
		user.dna.species.select_target(user)`